content/nic/installation/upgrade-version.md

```diff
@@ -0,0 +1,311 @@
+
+### Upgrade NGINX Ingress Controller charts
+
+Once the CRDs have been upgraded, you can then upgrade the release chart.
+
+The command depends on if you installed the chart using the registry or from source.
+
+To upgrade a release named _my-release_, use the following command:
+
+{{< tabs name="upgrade-chart" >}}
+
+{{% tab name="OCI registry" %}}
+
+```shell
+helm upgrade my-release oci://ghcr.io/nginx/charts/nginx-ingress --version {{< nic-helm-version >}}
+```
+
+{{% /tab %}}
+
+{{% tab name="Source" %}}
+
+```shell
+helm upgrade my-release .
+```
+
+{{% /tab %}}
+
+{{< /tabs >}}
+
+## Upgrade from 3.x to 4.x
+
+{{< call-out "warning" "This upgrade path is intended for 3.x to 4.0.0 only" >}}
+
+The instructions in this section are intended only for users upgrading from NGINX Ingress Controller 3.x to 4.0.0. Internal changes meant that backwards compability was not possible, requiring extra steps to upgrade.
+
+{{< /call-out >}}
+
+This section provides step-by-step instructions for upgrading NGINX Ingress Controller from version v3.x to v4.0.0.
+
+There are two necessary steps required: updating the `apiVersion` value of custom resources and configuring structured logging.
+
+For NGINX Plus users, there is a third step to create a Secret for your license.
+
+### Update custom resource apiVersion
+
+If the Helm chart you have been using is `v2.x`, before upgrading to NGINX Ingress Controller 4.0.0 you must update your GlobalConfiguration, Policy and TransportServer resources from `apiVersion: k8s.nginx.org/v1alpha1` to `apiVersion: k8s.nginx.org/v1`.
+
+If the Helm chart you have been using is `v1.0.2` or earlier (NGINX Ingress Controller `v3.3.2`), upgrade to Helm chart `v1.4.2` (NGINX Ingress Controller `v3.7.2`) before updating your GlobalConfiguration, Policy and TransportServer resources.
+
+The example below shows the change for a Policy resource: you must do the same for all GlobalConfiguration and TransportServer resources.
+
+{{<tabs name="resource-version-update">}}
+
+{{% comment %}} Keep this left aligned. {{% /comment %}}
+{{%tab name="Before"%}}
+
+```yaml
+apiVersion: k8s.nginx.org/v1alpha1
+kind: Policy
+metadata:
+  name: rate-limit-policy
+spec:
+  rateLimit:
+    rate: 1r/s
+    key: ${binary_remote_addr}
+    zoneSize: 10M
+```
+{{% /tab %}}
+
+{{%tab name="After"%}}
+```yaml
+apiVersion: k8s.nginx.org/v1
+kind: Policy
+metadata:
+  name: rate-limit-policy
+spec:
+  rateLimit:
+    rate: 1r/s
+    key: ${binary_remote_addr}
+    zoneSize: 10M
+```
+{{% /tab %}}
+
+{{</tabs>}}
+
+{{< warning >}}
+If a *GlobalConfiguration*, *Policy* or *TransportServer* resource is deployed with `apiVersion: k8s.nginx.org/v1alpha1`, it will be **deleted** during the upgrade process.
+{{</ warning >}}
+
+Once above specified custom resources are moved to `v1` ,please run below `kubectl` commands before upgrading to v4.0.0 Custom Resource Definitions (CRDs) to avoid [this issue](https://github.com/nginx/kubernetes-ingress/issues/7010).
```

Comment:
Thought it might be helpful to describe the issue that's linked to.

```suggestion
After you move the custom resources to `v1`, run the following `kubectl` commands before upgrading to v4.0.0 Custom Resource Definitions (CRDs) to avoid webhook errors caused by leftover `v1alpha1` resources. For details, see [GitHub issue #7010](https://github.com/nginx/kubernetes-ingress/issues/7010).
```

Reply:
active voice